### PR TITLE
feat(explain): add workspaces support

### PR DESCRIPTION
**Pull request by ruyadorno**
- Add highlight style for workspaces items in human output
- Add ability to filter results by workspace using `-w` config
- Added tests and docs

## References
Fixes: https://github.com/npm/statusboard/issues/300

### CODE REVIEW

File: lib/utils/explain-dep.js [hunk 1]
```diff
@@ -88,18 +107,23 @@ const explainDependents = ({ name, dependents }, depth, color) => {
 
 const explainEdge = ({ name, type, bundled, from, spec }, depth, color) => {
   const { bold } = color ? chalk : nocolor
+  const dep = type === 'workspace'
+    ? bold(relative('.', spec.slice('file:'.length)))
```
Comment (isaacs) [inline lib/utils/explain-dep.js]:
```suggestion
    ? bold(relative(from.location, spec.slice('file:'.length)))
```

This really needs to be a specific path, I think, not `.`, or else the tests are dependent on the working directory that they're run in.

(updated suggestion)

**Review by isaacs — CHANGES_REQUESTED** [review]
One slight change to make it not dependent on `cwd`, which ought to make the tests pass.

Other than that, looks good!

File: lib/utils/explain-dep.js (hunk 1 above)
Comment (isaacs) [reply lib/utils/explain-dep.js]: Oh, right, but edge is not an actual Arborist.Edge, it's a `node.explain().dependents` entry.  🤔

File: test/lib/utils/explain-dep.js [hunk 2]
```diff
@@ -1,8 +1,16 @@
+const { resolve } = require('path')
 const t = require('tap')
-const npm = {}
-const { explainNode, printNode } = t.mock('../../../lib/utils/explain-dep.js', {
-  '../../../lib/npm.js': npm,
-})
+const { explainNode, printNode } = require('../../../lib/utils/explain-dep.js')
+const testdir = t.testdir({})
```
Comment (isaacs) [inline test/lib/utils/explain-dep.js]: Could just use the `t.testdirName` getter here, if you're not actually using the folder.

Comment (ruyadorno) [reply test/lib/utils/explain-dep.js]: oh nice! didn't knew about that one 😄

Comment (ruyadorno) [reply test/lib/utils/explain-dep.js]: oops, is that _really_ a thing?

```
<path>/main/test/lib/utils/explain-dep.js:4
const testdir = t.testdirName()
                  ^
TypeError: t.testdirName is not a function
    at Object.<anonymous> (<path>/main/test/lib/utils/explain-dep.js:4:19)
    at Module._compile (node:internal/modules/cjs/loader:1109:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1138:10)
    at Module.load (node:internal/modules/cjs/loader:989:32)
    at Function.Module._load (node:internal/modules/cjs/loader:829:14)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12)
    at node:internal/main/run_main_module:17:47
```

Comment (isaacs) [reply test/lib/utils/explain-dep.js]: It's a string, not a function.

File: lib/utils/explain-dep.js (hunk 1 above)
Comment (isaacs) [reply lib/utils/explain-dep.js]: This works:

```js
bold(relative(from.location, spec.slice('file:'.length)))
```

File: test/lib/utils/explain-dep.js (hunk 2 above)
Comment (ruyadorno) [reply test/lib/utils/explain-dep.js]: aaaah, that makes sense then!

File: lib/utils/explain-dep.js (hunk 1 above)
Comment (ruyadorno) [reply lib/utils/explain-dep.js]: 💯

**Pull request merged**